src/main/java/mate/academy/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,52 @@
+package mate.academy.service.impl;
+
+import java.util.Collections;
+import java.util.Optional;
+import mate.academy.dao.ShoppingCartDao;
+import mate.academy.dao.TicketDao;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.MovieSession;
+import mate.academy.model.ShoppingCart;
+import mate.academy.model.Ticket;
+import mate.academy.model.User;
+import mate.academy.service.ShoppingCartService;
+
+@Service
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    @Inject
+    private ShoppingCartDao shoppingCartDao;
+    @Inject
+    private TicketDao ticketDao;
+
+    @Override
+    public void addSession(MovieSession movieSession, User user) {
+        Ticket ticket = new Ticket();
+        ticket.setMovieSession(movieSession);
+        ticket.setUser(user);
+        ticketDao.add(ticket);
+    }
+
+    @Override
+    public ShoppingCart getByUser(User user) {
+        Optional<ShoppingCart> optShoppingCart = shoppingCartDao.getByUser(user);
+        if (optShoppingCart.isEmpty()) {
+            throw new RuntimeException("Couldn't find shopping cart for user: " + user);
+        }
+        return optShoppingCart.get();
+    }
+
+    @Override
+    public void registerNewShoppingCart(User user) {
+        ShoppingCart shoppingCart = new ShoppingCart();
+        shoppingCart.setUser(user);
+        shoppingCart.setTickets(Collections.emptyList());
+        shoppingCartDao.add(shoppingCart);
```

Comment:
```suggestion
        shoppingCart.setUser(user);
        shoppingCartDao.add(shoppingCart);
```

I dont think we need this